earth_enterprise/docs/developer/BuildSystem.md

```diff
@@ -0,0 +1,130 @@
+
+* SCons
+* Gradle
+* Python scripts
+
+The code and bulid scripts for the different software products (Server,
+Fusion, Portable Globe) are not modularized, and the build script for one
+may reference the source code file structure for another.
+
+## Building Binary Utilities and Libraries
+
+### Server and Fusion Binaries
+
+There's virtually no separation between the scripts for building Server and
+Fusion.
+
+This process is performed by running `scons build` in the `earth_enterprise`
+directory.  It uses the SCons build tool, and has a few components of its own
+than need to be built before compiling Open GEE source code:
+
+* Bundled libraries
+* Generated source code
+* Fusion and Server library and executable binaries
+
+The SCons scripts for these components use a class named `khEnvironment` that
+derives from the SCons built-in `Environment`, and adds convenience methods.
+
+Server and Fusion can be built with different options.  There is a build
+directory name that's derived from the build options specified (e.g.,
+`NATIVE-REL-x86_64` or `NATIVE-DBG-x86_64`), and is created in
+`earth_enterprise/src` by default.
+
+This build directory has some notable subdirectories:
+
+* `bin` where built binaries are output
+* `bin/tests` where built unit and other test binaries are output
+* `lib` where built shared objects are output, and where shared objects
+  are looked for during compilation
+* `include` where include from built libraries are copied, and where header
+  files are looked for during compilation
+
+#### Bundled Libraries
+
+Each bundled library has files in a few places:
+
+* `earth_enterprise/third_party/<library-name>/<source-code-archive>.tar.gz`
+* `earth_enterprise/third_party/<library-name>/LICENSE`
+* `earth_enterprise/src/third_party/<library-name>/SConscript`
+
+There may be other files along the source code archive, such as a README file,
+patch files that are applied to the source code in the archive before
+building, obsolete `.spec` files for building RPM packages for the library.
+
+The `SConscript` file contains the build script that unpacks the library
+source code archive, applies patches, builds the library, and copies built
+binaries and other possible resources to the appropriated directories under
+the current build directory.
+
+All bundled libraries are built by
+`earth_enterprise/src/third_party/SConscript` in an order in which each
+library depends only on libraries listed before it in the build list.
+
+#### Generated Source Code
+
+Source code can be generated by Perl scripts, such as `AssetGenDaemonH.pl` and
+`AssetGenDaemonCpp.pl`.  The Perl commands are usually run from the same
+`SConscript` that compiles the generated source code and links it into a
+library, or an executable.
+
+#### Library and Executable Binaries
+
+Library and executable binaries for Fusion and Server are compiled and linked
+by `SConscripts` in the various directories under `src`.
```

Comment:
"... by `SConscript` files in..."

Reply:
`SConscript`s are `scripts`, so I think it's fine if they compile and link binaries.